retry/retry_example_test.go

```diff
@@ -0,0 +1,135 @@
+package retry_test
+
+// Go testable examples.
+// Any function in a test package with prefix "Example" is a "testable example".
+// It will be run as a test and the output must match the "Output:" in the comment.
+// See https://go.dev/blog/examples.
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/Pix4D/cogito/retry"
+)
+
+func ExampleRetry() {
+	rtr := retry.Retry{
+		UpTo:         5 * time.Second,
+		FirstDelay:   1 * time.Second,
+		BackoffLimit: 1 * time.Second,
+		Log: slog.New(slog.NewTextHandler(os.Stdout,
+			&slog.HandlerOptions{ReplaceAttr: removeTime})),
+	}
+
+	workFn := func() error {
+		// Do work...
+		return nil
+	}
+	classifierFn := func(err error) retry.Action {
+		if err != nil {
```

Comment:
is this needed in this example? How can you enter here in this specific example?

Reply:
Ah, I think I understand what you mean. You can enter here if in line 28 above, `// Do work...`, something fails and we return an error. I can make it clearer there.

Reply:
Yes, but since the function `workFn` here always returns  `nil` you can't enter this if statement in this case...

Reply:
Maybe I was not clear:

A comment of the type `// Do work...` is a placeholder for real code. I now extended that comment to make it clearer that it can return an error. I cannot really put a `return err` because it would break the test.

The fact that the literal code in the example does not contain a `return err`, does not imply that the classifier should not check for an error. The classifier MUST check for errors, it is the whole reason of existence of the classifier function...

Now, if this is not clear to you as a reader, it means that I must change the example. I can make it so it returns an error the first time or something similar. Do you feel that we need something like that?

Reply:
> The fact that the literal code in the example does not contain a return err

yes this was the confusing part: literal code missing this

> I can make it so it returns an error the first time or something similar.

Personally, I think it would make it clearer, but do as you think its best. Probably adding that comment is enough to clarify. On first read, my expectation was to see the literal code, since that was not there, I found it confusing later why we are checking for something that can't be achieved.

Reply:
I will have a look at this, but after the release.